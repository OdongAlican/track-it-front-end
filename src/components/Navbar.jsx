import React, { useState } from 'react'
import Signout from './auth/Signout'
import {useTransition, animated} from 'react-spring'


const Navbar = () => {
    const [showMenu, setShowMenu] = useState(false)

    const maskTransitions = useTransition(showMenu, null, {
            from: { position: 'absolute', opacity: 0.8 },
            enter: { opacity: 0.8 },
            leave: { opacity: 0.8 },
        })
    const menuTransitions = useTransition(showMenu, null, {
            from: { position: 0, transform: "translateX(-100%)" },
            enter: { opacity: 1, transform: "translateX(0%)" },
            leave: { opacity: 0, transform: "translateX(-100%)" },
        })

    return(
        <div className="d-flex pt-4 text-white nav-bar-section">
            <div className="header-section">
                <h3>Track it</h3>
            </div>
            <div className="itag-section">
                <i className="fas fa-bars"
                onClick={()=> setShowMenu(!showMenu)}
                ></i>
            </div>
            {
                maskTransitions.map(({ item, key, props }) =>
                item && 
                <animated.div 
                className="menu-mask"
                onClick={()=> setShowMenu(false)}
                key={key} 
                style={props}>
                </animated.div>
                )
            }
                        {
                menuTransitions.map(({ item, key, props }) =>
                item && 
                <animated.div 
                className="menu-section"
                key={key} 
                style={props}>
                    <Signout/>
                </animated.div>
                )
            }
        </div>
    )
}

export default Navbar