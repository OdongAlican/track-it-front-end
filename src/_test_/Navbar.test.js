import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar />, div);
});

it('renders nav-bar correctly', () => {
  const { getAllByTestId } = render(<Navbar />);
  const result = getAllByTestId('navbar');
  expect(result[0]).toHaveTextContent('Track it');
});

it('renders nav-bar correctly', () => {
  const { getAllByTestId } = render(<Navbar />);
  const result = getAllByTestId('navbar');
  expect(result[0]).not.toHaveTextContent('Track all');
});

it('matches navbar snapshot', () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
