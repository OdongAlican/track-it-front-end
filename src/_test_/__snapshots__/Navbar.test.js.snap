// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches navbar snapshot 1`] = `
<div
  className="d-flex pt-4 text-white nav-bar-section"
  data-testid="navbar"
>
  <div
    className="header-section"
  >
    <h3>
      Track it
    </h3>
  </div>
  <div
    className="itag-section"
  >
    <i
      className="fas fa-power-off"
      onClick={[Function]}
    />
  </div>
</div>
`;
